ext/dom/element.c

```diff
@@ -1239,20 +1238,19 @@ Since: DOM Level 2
 */
 static void dom_element_get_elements_by_tag_name_ns(INTERNAL_FUNCTION_PARAMETERS, bool modern)
 {
-	size_t uri_len, name_len;
 	dom_object *intern, *namednode;
-	char *uri, *name;
+	zend_string *uri, *name;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "p!p", &uri, &uri_len, &name, &name_len) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "P!P", &uri, &name) == FAILURE) {
 		RETURN_THROWS();
 	}
 
-	if (uri_len > INT_MAX) {
+	if (uri && ZSTR_LEN(uri) > INT_MAX) {
 		zend_argument_value_error(1, "is too long");
 		RETURN_THROWS();
 	}
```

Comment:
*might* make sense to define `uri` to the empty zend string instead of the ternary below?